app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -7,34 +7,110 @@ module DynamicGraphs.GraphGenerator
+                                      , counter = (counter stmtcounter) + 1}
+
+
+
+
+reqToStmts (name, J string1) stmtcounter = StmtsWithCounter{stmtslst = connectLst [DE $ DotEdge str1 name []] (stmtslst (foldUpReqLst [(name, NONE), (str1, NONE)] 0))
+                                                           , counter = (counter stmtcounter) + 3}
+  where str1 = pack string1
+
+--reqToStmts (name, AND reqs1) stmtcounter = connectLst [DN $ DotNode "and" [], DE $ DotEdge "and" name []] (connectLst andedges nodestatements)
+--  where andedges = createAndEdge (dragNodeIdInReq reqs1)
+--       nodestatements = concatMap reqToStmts ((name, NONE):(decompJString reqs1))
+
+--reqToStmts (name, OR reqs1) stmtcounter = connectLst [DN $ DotNode "or" [], DE $ DotEdge "or" name []] (connectLst oredges nodestatements)
+--  where oredges = createOrEdge (dragNodeIdInReq reqs1)
+--        nodestatements = concatMap reqToStmts ((name, NONE):(decompJString reqs1))
+
+--reqToStmts (name, FROM string1 (J string2)) stmtcounter = connectLst fromedges (concatMap reqToStmts [(name, NONE), (pack_string1, NONE), (pack_string2, NONE)])
+--  where pack_string1 = pack string1
+--        pack_string2 = pack string2
+--        fromedges = [DE $ DotEdge pack_string2 pack_string1 [], DE $ DotEdge pack_string1 name []]
+
+--reqToStmts (name, GRADE string1 (J string2)) stmtcounter = connectLst gradeedges (concatMap reqToStmts [(name, NONE), (pack string1, NONE), (pack string2, NONE)])
+--  where pack_string1 = pack string1
+--        pack_string2 = pack string2
+--        gradeedges = [DE $ DotEdge pack_string2 pack_string1 [], DE $ DotEdge pack_string1 name []]
+
+--reqToStmts (name, RAW string1) stmtcounter = connectLst rawedges (concatMap reqToStmts [(name, NONE), (pack_string1, NONE)])
+--  where pack_string1 = pack string1
+--        rawedges = [DE $ DotEdge pack_string1 name []]
+
+--reqToStmts (_, _) stmtcounter = []
+
+data StmtsWithCounter = StmtsWithCounter{ stmtslst :: [DotStatement Text]
+                                        , counter   :: Int }
+
+foldUpReqLst :: [(Text, Req)] -> Int -> StmtsWithCounter
+foldUpReqLst reqlst count = foldl(\acc x -> reqToStmts x acc) StmtsWithCounter{stmtslst = [], counter = count} reqlst
+
+
+
+
+decompJString :: [Req] -> [(Text, Req)]
+decompJString [] = []
+decompJString ((J x):xs) = (pack x, NONE):(decompJString xs)
+
+createAndEdge :: [Text] -> [DotStatement Text]
+createAndEdge [] = []
+createAndEdge (x:xs) = connectLst [DE $ DotEdge x "and" []] (createAndEdge xs)
+
+createOrEdge :: [Text] -> [DotStatement Text]
+createOrEdge [] = []
+createOrEdge (x:xs) = connectLst [DE $ DotEdge x "or" []] (createOrEdge xs)
+
+dragNodeIdInReq :: [Req] -> [Text]
+dragNodeIdInReq [] = []
+dragNodeIdInReq ((J text1):xs) = (pack text1):(dragNodeIdInReq xs)
+
+connectLst :: [a] -> [a] -> [a]
```

Comment:
This function is redundant; you can concatenate two lists by doing `xs ++ ys`.
While this function seems to reverse the first list, I'm not sure that's intentional or necessary.
Please remove this function.

Reply:
Oh, this function actually inserts the first element of xs at the front of ys, and then concatenate the rest of xs along with the new ys, so you still want to delete the function and do the procedures other ways?

Reply:
As I said, "While this function seems to reverse the first list, I'm not sure that's intentional or necessary." I have already described what this function is doing and said this is not necessary.